File: php/src/Domain/Topology/Entity/ZoneType.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Domain\Topology\Entity;

use Alxarafe\App\Domain\Topology\ValueObject\ZoneTypeCode;
use Alxarafe\App\Domain\Topology\ValueObject\ZoneTypeId;

/**
 * Master entity representing the type of a warehouse zone.
 *
 * Defines operational characteristics like whether the zone
 * is operative and whether it allows multi-SKU storage.
 */
final readonly class ZoneType
{
    public function __construct(
        private ZoneTypeId $id,
        private ZoneTypeCode $code,
        private bool $isOperative,
        private bool $allowsMultiSku,
    ) {
    }

    public function id(): ZoneTypeId
    {
        return $this->id;
    }

    public function code(): ZoneTypeCode
    {
        return $this->code;
    }

    public function isOperative(): bool
    {
        return $this->isOperative;
    }

    public function allowsMultiSku(): bool
    {
        return $this->allowsMultiSku;
    }
}

File: java/src/main/java/com/alxarafe/app/domain/topology/entity/ZoneType.java
package com.alxarafe.app.domain.topology.entity;

import com.alxarafe.app.domain.topology.valueobject.ZoneTypeCode;
import com.alxarafe.app.domain.topology.valueobject.ZoneTypeId;

import java.util.Objects;

/**
 * Master entity representing the type of a warehouse zone.
 *
 * Defines operational characteristics like whether the zone
 * is operative and whether it allows multi-SKU storage.
 */
public final class ZoneType {

    private final ZoneTypeId id;
    private final ZoneTypeCode code;
    private final boolean isOperative;
    private final boolean allowsMultiSku;

    public ZoneType(ZoneTypeId id, ZoneTypeCode code, boolean isOperative, boolean allowsMultiSku) {
        this.id = Objects.requireNonNull(id, "ZoneType id cannot be null.");
        this.code = Objects.requireNonNull(code, "ZoneType code cannot be null.");
        this.isOperative = isOperative;
        this.allowsMultiSku = allowsMultiSku;
    }

    public ZoneTypeId id() {
        return id;
    }

    public ZoneTypeCode code() {
        return code;
    }

    public boolean isOperative() {
        return isOperative;
    }

    public boolean allowsMultiSku() {
        return allowsMultiSku;
    }
}
